homeassistant/components/sensor/forecast.py

```diff
@@ -124,80 +116,66 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
+        self.forecast_data.update_unit_of_measurement(self.type)
```

Comment:
As I understand it, no. Unit of measure is not likely to ever change for the running system.